include/flexflow/model.h

```diff
@@ -856,10 +884,10 @@ class FFModel {
   std::unordered_map<size_t, NoOp*> cached_noop_ops;
   std::unordered_map<size_t, NoOp*> cached_input_ops;
   std::unordered_map<size_t, Cast*> cached_cast_ops;
-  std::unordered_map<size_t, Concat*> cached_concat_ops;
+  std::unordered_map<std::pair<ParallelTensorShapes, ConcatParams>, Concat*> cached_concat_ops;
   std::unordered_map<std::pair<ParallelTensorShape, Conv2DParams>, Conv2D*> cached_conv2d_ops;
   std::unordered_map<size_t, Dropout*> cached_dropout_ops;
-  std::unordered_map<size_t, ElementBinary*> cached_element_binary_ops;
+  std::unordered_map<std::pair<ParallelTensorShapes, ElementBinaryParams>, ElementBinary*> cached_element_binary_ops;
```

Comment:
What do you think about changing to either
```suggestion
  std::unordered_map<std::pair<std::pair<ParallelTensorShape, ParallelTensorShape>, ElementBinaryParams>, ElementBinary*> cached_element_binary_ops;
```
or 
```suggestion
  std::unordered_map<std::tuple<ParallelTensorShape, ParallelTensorShape, ElementBinaryParams>, ElementBinary*> cached_element_binary_ops;
```